kythe/go/indexer/testdata/code/methdecl.go

```diff
@@ -65,3 +65,145 @@ type w struct{}
+//- CtxName child.0 CtxCtx
+//- CtxCtx.kind "CONTEXT"
+//- CtxCtx child.0 CtxPkg
+//- CtxCtx child.1 CtxRec
+//- CtxCtx child.2 CtxFun
+//- CtxName child.1 CtxId
+//- CtxPkg.kind "IDENTIFIER"
+//- CtxPkg.pre_text "methdecl"
+//- CtxRec.kind "IDENTIFIER"
+//- CtxRec.pre_text "decorCommand"
+//- CtxFun.kind "IDENTIFIER"
+//- CtxFun.pre_text "Run"
+//- CtxId.kind "IDENTIFIER"
+//- CtxId.pre_text "ctx"
+//- CtxType.kind "LOOKUP_BY_TYPED"
+//- CtxType.lookup_index 0
+//-
+//- @Context ref CtxTypeValue
+//- CtxTypeValue code CtxTypeValueCode
+//- CtxTypeValueCode child.0 CtxTypeValueCodeCtx
+//- CtxTypeValueCodeCtx.kind "CONTEXT"
+//- CtxTypeValueCodeCtx.post_child_text "."
+//- CtxTypeValueCodeCtx child.0 CtxTypeValueCodeCtxChild
+//- CtxTypeValueCodeCtxChild.kind "IDENTIFIER"
+//- CtxTypeValueCodeCtxChild.pre_text "methdecl"
+//- CtxTypeValueCode child.1 CtxTypeValueCodeID
+//- CtxTypeValueCodeID.kind "IDENTIFIER"
+//- CtxTypeValueCodeID.pre_text "Context"
+//-
+//- @flag defines/binding RFFlag
+//- RFFlag code FlagCode
+//- FlagCode child.0 FlagName
+//- FlagName child.0 FlagCtx
+//- FlagCtx.kind "CONTEXT"
+//- FlagCtx child.0 FlagPkg
+//- FlagCtx child.1 FlagRec
+//- FlagCtx child.2 FlagFun
+//- FlagName child.1 FlagId
+//- FlagPkg.kind "IDENTIFIER"
+//- FlagPkg.pre_text "methdecl"
+//- FlagRec.kind "IDENTIFIER"
+//- FlagRec.pre_text "decorCommand"
+//- FlagFun.kind "IDENTIFIER"
+//- FlagFun.pre_text "Run"
+//- FlagId.kind "IDENTIFIER"
+//- FlagId.pre_text "flag"
+//-
+//- @FlagSet ref FlagTypeValue
+//- FlagTypeValue code FlagTypeValueCode
+//- FlagTypeValueCode child.0 FlagTypeValueCodeCtx
+//- FlagTypeValueCode child.1 FlagTypeValueCodeID
+//- FlagTypeValueCodeCtx.kind "CONTEXT"
+//- FlagTypeValueCodeCtx.post_child_text "."
+//- FlagTypeValueCodeCtx child.0 FlagTypeValueCodeCtxChild
+//- FlagTypeValueCodeCtxChild.kind "IDENTIFIER"
+//- FlagTypeValueCodeCtxChild.pre_text "methdecl"
+//- FlagTypeValueCodeID.kind "IDENTIFIER"
+//- FlagTypeValueCodeID.pre_text "FlagSet"
+//-
+//- @api defines/binding RFApi
+//- RFApi code ApiCode
+//- ApiCode child.0 ApiName
+//- ApiName child.0 ApiCtx
+//- ApiCtx.kind "CONTEXT"
+//- ApiCtx child.0 ApiPkg
+//- ApiCtx child.1 ApiRec
+//- ApiCtx child.2 ApiFun
+//- ApiName child.1 ApiId
+//- ApiPkg.kind "IDENTIFIER"
+//- ApiPkg.pre_text "methdecl"
+//- ApiRec.kind "IDENTIFIER"
+//- ApiRec.pre_text "decorCommand"
+//- ApiFun.kind "IDENTIFIER"
+//- ApiFun.pre_text "Run"
+//- ApiId.kind "IDENTIFIER"
+//- ApiId.pre_text "api"
+//-
+//- @API ref ApiTypeValue
+//- ApiTypeValue code ApiTypeValueCode
+//- ApiTypeValueCode child.0 ApiTypeValueCodeCtx
+//- ApiTypeValueCode child.1 ApiTypeValueCodeID
+//- ApiTypeValueCodeCtx.kind "CONTEXT"
+//- ApiTypeValueCodeCtx.post_child_text "."
+//- ApiTypeValueCodeCtx child.0 ApiTypeValueCodeCtxChild
+//- ApiTypeValueCodeCtxChild.kind "IDENTIFIER"
+//- ApiTypeValueCodeCtxChild.pre_text "methdecl"
+//- ApiTypeValueCodeID.kind "IDENTIFIER"
+//- ApiTypeValueCodeID.pre_text "API"
```

Comment:
This all makes me wonder if we had an s-exp (or other) representation for marked text literals that we could use, e.g.

```
  ApiTypeValue code '(children: (kind CONTEXT) (post: ".") (children ((kind IDENT) (pre "methdecl")) ((kind IDENT) (pre "API")))'
```

Reply:
I agree that having an easier way to record marked source assertions would be nice. I filed #4765.